docs/modules/ROOT/pages/access.adoc

```diff
@@ -133,6 +133,239 @@ there is no direct way for a contract to call circuits of other contracts or tra
+unlinkability, and collision resistance across deployments and ownership transfers.
+
+==== Owner ID Computation
+
+The foundation of the system is the owner identifier, computed as:
+
+```ts
+id = SHA256(pk, nonce)
+```
+
+Where `pk` is the owner's public key and `nonce` is a secret value that may be either randomly generated
+for maximum privacy or deterministically derived for recoverability.
+This identifier serves as a privacy-preserving alternative to exposing the raw public key,
+ensuring the owner's identity remains confidential.
+
+==== Owner Commitment Computation
+
+The final ownership commitment stored on-chain is computed as:
+
+```ts
+commitment = SHA256(id, instanceSalt, counter, pad(32, "ZOwnablePK:shield:"))
+```
+
+This multi-element hash provides several security properties:
+
+- `id`: The privacy-preserving owner identifier described above.
+- `instanceSalt`: A unique per-deployment salt that prevents commitment collisions across different contract instances, even when the same owner and nonce are used.
+- `counter`: Incremented with each ownership transfer to ensure unlinkability—each transfer produces a completely different commitment even with the same underlying owner.
+- `pad(32, "ZOwnablePK:shield:")`: A domain separator padded to 32 bytes that prevents hash collisions with other commitment schemes and enables safe protocol extensions.
+
+==== Security Properties
+
+This commitment scheme ensures that:
+
+- Public keys are never revealed on-chain.
+- Observers cannot correlate past and future ownership.
+- Cross-contract collisions are prevented through instance-specific salting.
+
+=== Nonce Generation Strategies
+
+The choice of nonce generation strategy represents a fundamental trade-off between simplicity/security and recoverability.
+Both approaches are valid, and the best choice depends on your specific threat model and operational requirements.
+
+==== Random Nonce
+
+Generating a cryptographically strong random nonce provides the strongest privacy guarantees:
+
+```typescript
+const randomNonce = crypto.getRandomValues(new Uint8Array(32));
+const ownerId = ZOwnablePK._computeOwnerId(publicKey, randomNonce);
+```
+
+This approach is easy to generate and ensures maximum unlinkability—even with sophisticated analysis,
+observers cannot correlate ownership across different contracts or time periods.
+However, it requires secure backup of both the private key and the nonce.
+*Loss of either component results in permanent, irrecoverable loss of ownership.*
+
+==== Deterministic Nonce
+
+Deriving the nonce deterministically enables recovery through derivation schemes.
+Some examples:
+
+- `H(passphrase + context)` - recoverable from passphrase only, but passphrase becomes critical single point of failure.
+- `H(publicKey + userPassphrase + context)` - requires both public key and passphrase.
+- `H(signature + context)` where `signature = sign(context)` - leverages wallet without exposing private key.
```

Comment:
Should we include a note or warning here that the user must ensure that the signature scheme used in the signature based recovery mechanisms must be deterministic? 

Fun fact: Ethereum's ECDSA implementation was non-deterministic prior to 2017.

Reply:
Mmm good point! On it

Reply:
[02227ab](https://github.com/OpenZeppelin/compact-contracts/pull/204/commits/02227ab2cfa624606d33227df575206008feea88)